app/models/piece.rb

```diff
@@ -1,3 +1,64 @@
 class Piece < ActiveRecord::Base
   belongs_to :game
+
```

Comment:
`Public` and `Private` methods.

It currently looks like the `obstructed?` method is the only one that needs to be public (used by other classes). The rest can be `private`. So it's always good practice to make internal methods private.

Here is a good blog post on it: http://culttt.com/2015/06/03/the-difference-between-public-protected-and-private-methods-in-ruby/